src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Entities/EntityExtensions.cs

```diff
@@ -56,14 +52,22 @@ public static bool Has<T>(this IEntity entity)
         /// <param name="name">The name of the property to check.</param>
         /// <returns>True if the property was found, otherwise false.</returns>
         public static bool Has(this IEntity entity, string name)
-            => entity.Properties[name] != null;
+            => name != null && entity.Properties.ContainsKey(name);
```

Comment:
Check if it's null or empty

Reply:
no need. we can store a key with empty string.

Reply:
This is impossible for more than one entry, right?

Reply:
Yes it's a dictionary.  Null throws an exception but empty string won't.